plugins/connectors/src/Connectors.cpp

```diff
@@ -171,36 +226,65 @@ namespace mars {
 
       }
 
-      void Connectors::checkForPossibleConnections() {
-        std::string malename, femalename, maletype, femaletype;
+      void Connectors::checkForPossibleConnections(bool isforced) {
+        // Connector properties.
+        std::string malename, femalename, maletype, femaletype, maleautoconnect, femaleautoconnect;
+
+        // Merge type to check against.
+        std::string automatic ("automatic");
+
+        // Nested for loop to check for every male-female connector mating combination.
         for (std::map<std::string, configmaps::ConfigMap>::iterator mit= maleconnectors.begin();
           mit!=maleconnectors.end(); ++mit) {
+
+          // Get male connector properties.
           malename = (std::string)mit->second["name"];
           maletype = (std::string)mit->second["type"];
+          maleautoconnect = (std::string)mit->second["mating"];
```

Comment:
I don't know of any other values for "mating". This is based on what existed in the YAML config for the payload, i.e. "mating: automatic" for both male and female connectors: https://github.com/rock-simulation/mars/pull/106/commits/7a3e6e8d2b9a50b4a4652bf7c7a2b7eea105a2d5

If it were up to me then it would be a boolean property that would also be called autoconnect as all it does is override whatever is the global autoconnect setting. However, because the use case of this mating property is not documented I left it as is in order to not risk breaking however it is used elsewhere. That being said, the mating property was not being used at all in the plugin so.... ¯\_(ツ)_/¯

Reply:
You mention three types of autoconnect.

In the lines above, the setting is not relating to any global connection setup but to the specific one of the male. Seems to me more clear to have maleconnectmode which can take any value given in the map. I mean,  naming it maleautoconnect seems to me, makes things just more complicated.  I think here it can only take the values automatic or else.

Unless there is a good reason to call it `maleautoconnect` I would stick to `malemating` or `maleconnectmode`. `malemating` is  good because it directly links with the key in the dictionary. `maleconnectmode` is not bad because `connectmode` and `mating` can be understood as synonyms but actually this can lead already to some confusion. 

Should we name the variable malemating? Otherwise, rename the dictionary key and the variable to `connectmode`

Am I missing something? I think that the general mating configuration is not directly influencing this discussion, right?

Reply:
Another option is to rename the key dictionary to `autoconnect` and make it boolean. Then we can keep the `maleautoconnect` variable as it is.

Reply:
Will rename to malemating and femalemating.